packages/ui/src/SelectField.tsx

```diff
@@ -36,22 +37,27 @@ export type SelectFieldOption<V = string> = {
   value: V
 }
 
+export type SelectFieldValue<V = string> = SelectFieldOption<V> | SelectFieldOption<V>[]
+
 export type SelectFieldCoreStaticProps = {
   name: string
   onBlur?: (e: FocusEvent<HTMLElement>) => void
   error?: string
+  isMulti?: boolean
 }
+
 export type SelectFieldCoreDynamicProps<V> =
   | {
       isClearable: true
-      value: SelectFieldOption<V> | null
-      onChange: (newValue: SelectFieldOption<V> | null) => void
+      value: SelectFieldValue<V> | null
```

Comment:
@utkukaratas I am not sure about this typing. Potentially, it allows users to pass value: V and the onChange that expects V[] back. It was one of the reasons I suggested to have MultiSelect as a separate component

Reply:
hmm well if a type-happy coder like you finds it confusing we better seperate it :)) incoming...

Reply:
Let's ask the team.

Reply:
(context: do we create a seperate MultiSelect component **VS** augment SelectField with more intricate types to handle multiple selections)

Reply:
The current type seems good enough for common use-cases, so I'd be fine with keeping that. I don't have a strong preference, though.

Reply:
Making almost identical component again because of this seems like an overkill to me. I know we want to be clean but maybe that's the case we're fine with. A comment might help if the user struggles with types.

Reply:
@utkukaratas @aigoncharov Maybe this is a solution?
```
export type SelectFieldCoreDynamicProps<V> =
  | {
      isClearable: true
      value: SelectFieldOption<V> | null
      onChange: (newValue: SelectFieldOption<V> | null) => void
    }
  | {
      isClearable?: false
      value: SelectFieldOption<V>
      onChange: (newValue: SelectFieldOption<V>) => void
    }
  | {
      isClearable: true
      value: SelectFieldOption<V>[] | null
      onChange: (newValue: SelectFieldOption<V>[] | null) => void
    }
  | {
      isClearable?: false
      value: SelectFieldOption<V>[]
      onChange: (newValue: SelectFieldOption<V>[]) => void
    }
```

Reply:
@danad02 good idea!

Reply:
that's a mouthful but why not. thx @danad02. on its way...